### PR TITLE
perf(stream,agg): reuse existing `count(*)` as internal row count agg call

### LINKED ISSUE #8197: opt(agg): reuse existing `count(*)` while generating stream plan

### PR DESCRIPTION
**Pull request by richardchien**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR makes `HashAggExecutor` and `GlobalSimpleExecutor` reuse existing `count(*)` written by user as their internal row count agg call. This will close #8197.

Incidentally, it also removes unneeded `count(*)` inserted into `LocalSimpleAgg`.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Review by st1page — COMMENTED** [review]
-1 for implementing the logic in the streamNode's construct function. because we can only add the StreamProject outside and the developer might forget it. It seems like a leaky abstraction. Maybe it is better to append the `count(*)` before constructing the StreamAgg and assert it exists in the construct function.

**Review by st1page — CHANGES_REQUESTED** [review]
seems there is something wrong on the merge checking and after I comment the PR, it reminds me it can be merged :hot_face:

File: src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs [hunk 1]
```diff
@@ -97,6 +97,7 @@ impl StreamNode for StreamLocalSimpleAgg {
                 .iter()
                 .map(PlanAggCall::to_protobuf)
                 .collect(),
+            row_count_index: u32::MAX, // this is not used
```
[the remark this answers is not in the document]
Comment (richardchien) [reply src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs:100]: +1

**Review by st1page — APPROVED** [review]
LGTM

File: src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (BugenZhao) [reply src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs:100]: What about documenting this explicitly in proto's comment?

File: src/frontend/src/optimizer/plan_node/stream_hash_agg.rs [hunk 2]
```diff
@@ -27,14 +27,23 @@ use crate::utils::{ColIndexMapping, ColIndexMappingRewriteExt};
...
+    vnode_col_idx: Option<usize>,
+
+    /// The index of `count(*)` in `agg_calls`.
+    row_count_idx: usize,
 }
 
 impl StreamHashAgg {
-    pub fn new(logical: LogicalAgg, vnode_col_idx: Option<usize>) -> Self {
+    pub fn new(logical: LogicalAgg, vnode_col_idx: Option<usize>, row_count_idx: usize) -> Self {
+        debug_assert_eq!(
```
Comment (BugenZhao) [inline src/frontend/src/optimizer/plan_node/stream_hash_agg.rs:42]: This is lightweight and it should be okay to remove the `debug_`.

**Review by BugenZhao — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/stream_hash_agg.rs (hunk 2 above)
Comment (richardchien) [reply src/frontend/src/optimizer/plan_node/stream_hash_agg.rs]: Ok

File: src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardchien) [reply src/frontend/src/optimizer/plan_node/stream_local_simple_agg.rs:100]: I guess it's better to introduce a new type of node. They are too different.🤣

**Pull request merged**